### PR TITLE
[Translation] merge all fallback catalogues messages into current catalo...

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | ~ |
| Tests pass? | yes |
| License | MIT |

The amount of reduced memory depend on number of defined fallbacks and the size of messages.
For example if we defined 2 fallbacks and we have a large file like ~20 mb we save (~40 mb) 
selection_004

### CODE REVIEW

File: src/Symfony/Component/Translation/Translator.php [hunk 1]
```diff
@@ -423,6 +401,52 @@ private function initializeCacheCatalogue($locale, $forceRefresh = false)
...
+            while ($fallbackCatalogue) {
+                $messages = array_replace_recursive($fallbackCatalogue->all(), $messages);
+                $fallbackCatalogue = $fallbackCatalogue->getFallbackCatalogue();
+            }
+            foreach ($messages as $domain => $domainMessages) {
+                $catalogue->add($domainMessages, $domain);
+            }
+        } else {
+            $current = '';
+            $fallbackContent = '';
```
Comment (cordoval) [inline src/Symfony/Component/Translation/Translator.php]: fallbackContent is already ''

File: src/Symfony/Component/Translation/Translator.php [hunk 2]
```diff
@@ -423,6 +401,52 @@ private function initializeCacheCatalogue($locale, $forceRefresh = false)
...
+            $fallbackCatalogue = $catalogue->getFallbackCatalogue();
+            while ($fallbackCatalogue) {
+                $fallback = $fallbackCatalogue->getLocale();
+                $fallbackSuffix = ucfirst(preg_replace($replacementPattern, '_', $fallback));
+                $currentSuffix = ucfirst(preg_replace($replacementPattern, '_', $current));
+
+                $fallbackContent .= sprintf(<<<EOF
+\$catalogue%s = new MessageCatalogue('%s', %s);
+\$catalogue%s->addFallbackCatalogue(\$catalogue%s);
+
```
Comment (cordoval) [inline src/Symfony/Component/Translation/Translator.php]: wondering why two line breaks?

File: src/Symfony/Component/Translation/Translator.php (hunk 1 above)
Comment (aitboudad) [inline src/Symfony/Component/Translation/Translator.php]: fixed

File: src/Symfony/Component/Translation/Translator.php [hunk 3]
```diff
@@ -423,6 +401,49 @@ private function initializeCacheCatalogue($locale, $forceRefresh = false)
...
+            // merge all fallback catalogues messages into $catalogue
+            $fallbackCatalogue = $catalogue->getFallbackCatalogue();
+            $messages = $catalogue->all();
+            while ($fallbackCatalogue) {
+                $messages = array_replace_recursive($fallbackCatalogue->all(), $messages);
+                $fallbackCatalogue = $fallbackCatalogue->getFallbackCatalogue();
+            }
+            foreach ($messages as $domain => $domainMessages) {
+                $catalogue->add($domainMessages, $domain);
+            }
```
Comment (fabpot) [inline src/Symfony/Component/Translation/Translator.php]: I would add `return '';` here and remove the `else`.

Comment (aitboudad) [inline src/Symfony/Component/Translation/Translator.php]: fixed

**Comment by fabpot** [comment]
Thank you @aitboudad.

**Pull request merged**